src/Agda2Hs/Compile/Record.hs

```diff
@@ -105,15 +107,12 @@ compileRecord target def = do
               assts  -> Just (Hs.CxTuple () assts)
         defaultDecls <- compileMinRecords def ms
         return $ Hs.ClassDecl () context hd [] (Just (classDecls ++ map (Hs.ClsDecl ()) defaultDecls))
-      ToRecord ds -> do
-        checkValidConName cName
-        (constraints, fieldDecls) <- compileRecFields fieldDecl recFields fieldTel
-        compileDataRecord constraints fieldDecls (Hs.DataType ()) hd ds
-      ToRecordNewType ds -> do
+      ToRecord newtyp ds -> do
         checkValidConName cName
         (constraints, fieldDecls) <- compileRecFields fieldDecl recFields fieldTel
-        checkSingleElement cName fieldDecls "Newtype must have exactly one field in constructor"
-        compileDataRecord constraints fieldDecls (Hs.NewType ()) hd ds
+        when newtyp $ checkSingleElement cName fieldDecls "Newtype must have exactly one field in constructor"
```

Comment:
There is some repetition here with `Data.checkNewtype`; can you factor this out so that we also keep consistent messages, e.g. by inlining `checkNewtype` or simply re-using a defined `singleConstructorField` message.